File: db/csharp/SalesManagement.Domain/Models/AutoNumber.cs
namespace SalesManagement.Domain.Models;

/// <summary>
/// 自動採番マスタ
/// </summary>
public class AutoNumber
{
    /// <summary>
    /// 伝票種別コード (OR:受注 SA:売上 PO:発注 PU:仕入 IN:請求 CR:入金 PA:支払)
    /// </summary>
    public string SlipType { get; set; } = string.Empty;

    /// <summary>
    /// 年月 (YYYYMM形式)
    /// </summary>
    public string YearMonth { get; set; } = string.Empty;

    /// <summary>
    /// 最終伝票番号
    /// </summary>
    public int LastSlipNo { get; set; }

    /// <summary>
    /// 作成日時
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 作成者名
    /// </summary>
    public string CreatedBy { get; set; } = string.Empty;

    /// <summary>
    /// 更新日時
    /// </summary>
    public DateTime UpdatedAt { get; set; }

    /// <summary>
    /// 更新者名
    /// </summary>
    public string UpdatedBy { get; set; } = string.Empty;
}

File: db/fsharp/SalesManagement.Domain/Models/AutoNumber.fs
namespace SalesManagement.Domain.Models

open System

/// 自動採番マスタ
[<CLIMutable>]
type AutoNumber = {
    SlipType: string             // 伝票種別コード（PK）(OR:受注 SA:売上 PO:発注 PU:仕入 IN:請求 CR:入金 PA:支払)
    YearMonth: string            // 年月（PK、YYYYMM形式）
    LastSlipNo: int              // 最終伝票番号
    CreatedAt: DateTime          // 作成日時
    CreatedBy: string option     // 作成者名
    UpdatedAt: DateTime          // 更新日時
    UpdatedBy: string option     // 更新者名
}
